modules/cache/main.tf

```diff
@@ -0,0 +1,66 @@
+data "aws_route53_zone" "public" {
+  name = var.zone_name
+}
+
+module "redis" {
+  count = var.cache_store == "redis" ? 1 : 0
+
+  source  = "cloudposse/elasticache-redis/aws"
+  version = "0.30.0"
+
+  name = "redis"
+
+  vpc_id               = var.vpc_id
+  namespace            = var.project
+  stage                = var.environment
+  subnets              = var.subnet_ids
+  availability_zones   = var.availability_zones
+  replication_group_id = "${var.project}-${var.environment}-redis"
+
+  family         = var.redis_family
+  engine_version = var.redis_engine_version
+  instance_type  = var.instance_type
+  cluster_size   = var.cluster_size
+
+  allowed_security_groups = var.security_groups
+
+  zone_id = data.aws_route53_zone.public.zone_id
+
+  cloudwatch_metric_alarms_enabled = true
+
+  transit_encryption_enabled = false
```

Comment:
Any reason this is disabled? I think we should either set it to true or have a var for it and default it to true.

Reply:
I prefer put this setting with false  by default, and do true next round.

When turns true, redis SSL is enabled. I would provide a solution together with stunnel gateway for k8s pod to access directly without SSL handling. I have tested the stunnel solution within a pod and the app can do localhost access using redis-cli.

Considering that most redis clients have built-in TLS support, we can leave true/false to user, but false by default, so that the app can chat with redis server directly via private network and securely.